### ISSUE TITLE
Feature: Convert kraus representation to channel (superoperator)

### ISSUE THREAD

**Issue by vprusso** [vprusso/toqito#14]
Given a collection of Kraus operators, convert them into the corresponding quantum channel (i.e. the superoperator). 

This would be placed under `channel_ops/kraus_to_channel.py`. Corresponding tests would also need to be created in the `tests/test_channel_ops/test_kraus_to_channel.py`file. Updating the docs would also need to be done as well.

**Comment by sebgrijalva** [comment]
I suppose that by the superoperator you mean the natural representation of the quantum channel (according to Watrous' book). So I guess that something like eq. 2.74, p 80. of Watrous' book would work here:

```python
 super_op = sum(toqito.matrix_ops.tensor(A, B.conj()) for A, B in kraus_list)
```

I tried this with some random Kraus operators

```python
dim = 2**2
kraus_list = [np.random.randint(-1, 4, (2, dim, dim)) for _ in range(12)]
```
and some random density matrix (together with its vectorization):
```python
vector = np.random.randint(-3, 3, (dim, 1))
dm = toqito.state_ops.pure_to_mixed(vector)
vec_dm = toqito.matrix_ops.vec(dm)
```
and compared the action of the superoperator on the (vectorized) density matrix `toqito.matrix_ops.unvec(superop @ vec_dm)
`, versus the action of the Kraus operators on the density matrix `sum(A @ dm @ B.conj().T for A, B in kraus_list)`. I get the same entries, but one matrix is the transpose of the other, so I was wondering if I'm getting all the operation definitions right. May I ask you to take a look at these definitions? There's probably a bug that I'm not seeing.

Thanks!

**Comment by vprusso** [comment]
Hi @sebgrijalva 

Thanks very much for your question, and do let me know if what I'm providing here makes sense.

> I suppose that by the superoperator you mean the natural representation of the quantum channel (according to Watrous' book). So I guess that something like eq. 2.74, p 80. of Watrous' book would work here:
> 
> ```python
>  super_op = sum(toqito.matrix_ops.tensor(A, B.conj()) for A, B in kraus_list)
> ```

Hmm, I believe that if your `kraus_list` that you're iterating over already contains the Kraus operators obtained elsewhere, these should already have the conjugation applied to the second right Kraus operator--in which case I believe the code would look something like:

```python
super_op = sum(toqito.matrix_ops.tensor(A, B) for A, B in kraus_list)
```

This may be responsible for why your matrices at the end have the same entries but are a transpose of the other.

**Comment by Shivansh20128** [comment]
Hi! I want to work on this if this is still relevant to the project. However, I must do a background study about Kraus operations before adding something. From what I know right now, you want to replace a list a Kraus operators with a single _superoperator_, and the approach discussed above makes sense to me. Let me know if I am on the right track and please feel free to assign this to me if this feature is still needed. 
Thank you

**Comment by vprusso** [comment]
Hi @Shivansh20128 ! 

Sounds great, and yes, you have the right idea! Don't hesitate to reach out with any questions you may have, and thank you again, as always, for your interest and great contributions to the `toqito` project!

**Comment by Shivansh20128** [comment]
Hi @vprusso , I implemented the function for kraus_to_channel as suggested by you in the above comments, but the problem mentioned still exists. One matrix is the transpose of the other. Can you take a look and see if this is what you had in mind?

<img width="948" alt="Screenshot 2024-10-19 at 11 58 46 AM" src="https://github.com/user-attachments/assets/e89f65e1-b7b7-4084-b54f-57e8b118d6ec">

**Comment by Shivansh20128** [comment]
Now that I thought about it, removing conjugation in the line `super_op = sum(toqito.matrix_ops.tensor(A, B.conj()) for A, B in kraus_list)` to convert `B.conj()` to `B` won't change anything because the numbers involved are purely real (have no complex part). 
I think the change required here has something to do with the matrix representations in the two methods. The tensor(A, B.conj()) method internally uses a different matrix representation, which often aligns with a row-major ordering versus the standard matrix multiplication, which follows the usual ordering. So we need to take the **Transpose** of one of the representations, but I am not sure which one.
What are you views on this?

**Comment by Shivansh20128** [comment]
Hi! The milestone is coming close🥲. Are we still working to achieve the issue we aimed for? I need a little help here. Can someone look at the above comments?
Thank you

**Comment by vprusso** [comment]
Yes, the milestone is indeed closing out. I'll aim for another release at the end of this month, and whatever is completed by that point would be great to include (including much of your great work!). 

As for help, do you have a branch with your changes pushed up that I can look at? That might be helpful for me to dig into what might be going on here.

**Comment by Shivansh20128** [comment]
> Yes, the milestone is indeed closing out. I'll aim for another release at the end of this month, and whatever is completed by that point would be great to include (including much of your great work!).

Okay cool.
> As for help, do you have a branch with your changes pushed up that I can look at? That might be helpful for me to dig into what might be going on here.

No, I haven't yet started working on this issue wrt to code changes. I am still in the research stage about what channel representation would work appropriately given the Kraus operators.

**Comment by purva-thakre** [comment]
@Shivansh20128 Planning to finish this issue?

**Comment by Shivansh20128** [comment]
Yes I was working on it, but then I didn't get any feedback on my [this](https://github.com/vprusso/toqito/issues/14#issuecomment-2423655307) comment. Maybe I can add a draft PR and show what I have done so far.

**Comment by purva-thakre** [comment]
> Maybe I can add a draft PR and show what I have done so far.

That would be better.
